fish-rust/src/builtins/string/unescape.rs

```diff
@@ -0,0 +1,54 @@
+use super::{Arguments, StringError, StringSubCommand};
+use crate::builtins::shared::{io_streams_t, STATUS_CMD_ERROR, STATUS_CMD_OK};
+use crate::common::{unescape_string, UnescapeStringStyle};
+use crate::ffi::parser_t;
+use crate::wchar::{wstr, L};
+use crate::wgetopt::{wopt, woption, woption_argument_t::*};
+
+#[derive(Default)]
+pub struct Unescape {
+    no_quoted: bool,
+    style: UnescapeStringStyle,
+}
+
+impl StringSubCommand<'_> for Unescape {
+    const LONG_OPTIONS: &'static [woption<'static>] = &[
+        wopt(L!("no-quoted"), no_argument, 'q'),
```

Comment:
no-quoted has short option 'n'. Distressing that this wasn't caught by the tests.

Reply:
I think this is rather a fault in that `no-quoted` does not mean anything for `unescape` — it did nothing and does nothing, the `NO_QUOTED` flag is not defined for unescapeflags https://github.com/fish-shell/fish-shell/blob/ade66505997cd38286923b6ab109814e1b580a27/fish-rust/src/common.rs#L134. It should probably rather be removed, but that is a potentially breaking change. 

It is also not documented that it should support `no-quoted` https://fishshell.com/docs/current/cmds/string-unescape.html#synopsis

Reply:
I fixed this so it the correct `n`, but left the option still there